.github/ensure-builder/action.yml

```diff
@@ -0,0 +1,215 @@
+    required: false
+  runner_type:
+    required: true
+  github_token:
+    required: true
+  aws_access_key_id:
+    required: true
+  aws_secret_access_key:
+    required: true
+  run_id:
+    required: false
+    # build instances need a folder per workflow run, testers can use default
+    default: "project"
+  run_attempt:
+    required: true
+  ec2_key:
+    required: true
+  concurrency_github_token:
+    required: true
+  run:
+    # command to run once builder/tester available
+    required: false
+runs:
+  # define an action, runs in OS of caller
+  using: composite
+  steps:
+    - name: Select Instance Type and AMI
+      id: select_instance
+      shell: bash
+      run: |
+        TYPE=${{ inputs.runner_type }}
+        echo "spot_strategy=None" >> $GITHUB_OUTPUT
+        if [[ $TYPE == builder-x86 ]]; then
+          # 128-core x86 instance types with least evictions
+          echo "instance_type=r6in.32xlarge r6a.32xlarge i4i.32xlarge r7iz.32xlarge" >> $GITHUB_OUTPUT
+          echo "ami_id=ami-04d8422a9ba4de80f" >> $GITHUB_OUTPUT
+          echo "ebs_cache_size=256" >> $GITHUB_OUTPUT
+          echo "runner_concurrency=50" >> $GITHUB_OUTPUT
+          echo "ttl=40" >> $GITHUB_OUTPUT
+        elif [[ $TYPE == builder-arm ]]; then
+          # 64-core ARM instance type
+          echo "instance_type=r6g.16xlarge" >> $GITHUB_OUTPUT
+          echo "ami_id=ami-0d8a9b0419ddb331a" >> $GITHUB_OUTPUT
+          echo "ebs_cache_size=256" >> $GITHUB_OUTPUT
+          echo "runner_concurrency=8" >> $GITHUB_OUTPUT
+          echo "ttl=40" >> $GITHUB_OUTPUT
+        elif [[ $TYPE == bb-bench-x86 ]]; then
+          # 16-core x86 instance type
+          echo "instance_type=m6a.4xlarge" >> $GITHUB_OUTPUT
+          echo "ami_id=ami-04d8422a9ba4de80f" >> $GITHUB_OUTPUT
+          echo "ebs_cache_size=64" >> $GITHUB_OUTPUT
+          echo "runner_concurrency=1" >> $GITHUB_OUTPUT
+          echo "ttl=40" >> $GITHUB_OUTPUT
+        else
+          echo "spot_strategy=BestEffort" >> $GITHUB_OUTPUT
+          if [[ $TYPE == *tester-x86* ]]; then
+            echo "ami_id=ami-04d8422a9ba4de80f" >> $GITHUB_OUTPUT
+            # no github runners, 'bare spot' in action code
+            echo "runner_concurrency=0" >> $GITHUB_OUTPUT
+            echo "ttl=20" >> $GITHUB_OUTPUT
+          fi
+          if [[ $TYPE == 4core-* ]]; then
+            echo "instance_type=m6a.2xlarge" >> $GITHUB_OUTPUT
+          elif [[ $TYPE == 8core-* ]]; then
+            echo "instance_type=m6a.4xlarge" >> $GITHUB_OUTPUT
+          elif [[ $TYPE == 16core-* ]]; then
+            echo "instance_type=m6a.8xlarge" >> $GITHUB_OUTPUT
+          elif [[ $TYPE == 32core-* ]]; then
+            echo "instance_type=m6a.16xlarge" >> $GITHUB_OUTPUT
+          elif [[ $TYPE == 64core-* ]]; then
+            echo "instance_type=m6a.32xlarge" >> $GITHUB_OUTPUT
+          elif [[ $TYPE == 128core-* ]]; then
+            echo "instance_type=m6a.64xlarge" >> $GITHUB_OUTPUT
```

Comment:
pretty sure 128 core (or vcpu but we use interchangeably), is 32x

Reply:
good catch